server/src/main/java/org/elasticsearch/common/io/stream/StreamInput.java

```diff
@@ -525,13 +536,14 @@ private static void throwOnBrokenChar(int c) throws IOException {
     }
 
     public SecureString readSecureString() throws IOException {
-        BytesReference bytesRef = readBytesReference();
-        byte[] bytes = BytesReference.toBytes(bytesRef);
-        try {
-            return new SecureString(CharArrays.utf8BytesToChars(bytes));
-        } finally {
-            Arrays.fill(bytes, (byte) 0);
```

Comment:
This is just as "safe" now as it was before. We never zeroed out the network buffer, now we also avoid copying here. Not really sure how this functionality makes any sense to be honest but this PR doesn't change behavior and saves some allocations here.